src/rabbitmq/rabbitmq.ts

```diff
@@ -0,0 +1,53 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import amqplib, { Channel } from 'amqplib/callback_api';
+import config from 'config';
+import { DataSourceDocument } from '../data-source.model';
+
+interface HarvestMessage {
+  orgId: string;
+  catalogId: string;
+  dataSourceType: string;
+}
+
+const { host = '', port = '', exchange = '' } = config.get('rabbitmq');
+
+let channel: Channel;
+
+amqplib.connect(`amqp://${host}:${port}`, (err, connection) => {
+  if (err) {
+    throw err;
+  }
+
+  connection.on('error', console.error);
+
+  connection.createChannel((err, ch) => {
+    if (err) {
+      throw err;
+    }
+
+    ch.assertExchange(exchange, 'fanout', { durable: false });
+    channel = ch;
+  });
+});
+
+export const publishDataSource = ({
+  dataSourceType = '',
+  publisherId = ''
+}: DataSourceDocument): void => {
+  const message: HarvestMessage = {
+    orgId: publisherId,
+    catalogId: publisherId,
+    dataSourceType: dataSourceType
```

Comment:
finne ut hvis det er i spec eller ikke, fjerne foreløpig - vi bruker message type/topic til å finne ut

Reply:
snakket med Stig og vi lar den heller være der foreløpig, så sletter vi den hvis vi finner ut at vi absolutt ikke trenger det

Reply:
jeg anbefaler at vi ikke lar konfliktmuligheter inni systmeet.  Typ er i meldingstyp eller i payload. Hvis du har begge må du ha ekstra if og errorhandling, og testing og testcoverage... best å ikke ha mulihet for konflikter